frontend/src/components/dashboard/settings/api-key/CreateModal.tsx

```diff
@@ -0,0 +1,69 @@
+import { useState } from 'react'
+
+import styles from './CreateModal.module.scss'
+
+import { PrimaryButton, TextInput } from 'components/common'
+import { ApiKey, generateApiKey } from 'services/api-key.service'
+
+export const CreateModal = ({
+  onSuccess,
+}: {
+  onSuccess: (key: ApiKey) => void | Promise<void>
+}) => {
+  const [keyLabel, setKeyLabel] = useState<string>('')
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [isError, setIsError] = useState<boolean>(false)
+  return (
+    <form
+      onSubmit={(e) => {
+        e.preventDefault()
+        if (isLoading) {
+          return
+        }
+
+        void (async () => {
+          try {
+            setIsLoading(true)
+            const apiKey = await generateApiKey({ label: keyLabel })
+            await onSuccess(apiKey)
+          } catch (e) {
+            setIsError(true)
+            setTimeout(() => setIsError(false), 5000)
+          } finally {
+            setIsLoading(false)
+          }
+        })()
+      }}
+    >
+      <div>
+        <h2>
+          <label htmlFor="apiKeyLabel">Create a new API key</label>
+        </h2>
+        <h4>Key Label</h4>
+        <p>
+          The label you choose for your API key is be permanent and cannot be
+          changed once it has been created.
+        </p>
+        <TextInput
+          id="apiKeyLabel"
+          type="text"
+          placeholder="The label that you can identify your API key with"
```

Comment:
**[nit]** Sorry missed this earlier, strictly grammatical English should be: `The label with which you can identify your API key`